common/metrics/otel_event_handler.go

```diff
@@ -90,18 +86,7 @@ func (m *OtelMetricHandler) Event(ctx context.Context, e *event.Event) context.C
 	return ctx
 }
 
-func (m *OtelMetricHandler) getRecordFunc(em event.Metric) recordFunc {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	if f, ok := m.recordFuncs[em]; ok {
```

Comment:
so event.Metric is unbounded? 

And instead of caching recordFuns, we create a new one every time? Will it impact latency or gc?

Reply:
The difference is the GC is now collecting what was leaking before.

Reply:
not sure I understand the unbounded, the objects needed in the function are captured, em is only needed for logging before returning recording function

Reply:
the events coming as parameters to the `Event` function are pooled, it is not unbounded.